controllers/reconcilers/grafana/grafana_service_reconciler.go

```diff
@@ -48,7 +48,12 @@ func (r *ServiceReconciler) Reconcile(ctx context.Context, cr *v1beta1.Grafana,
 
 	// try to assign the admin url
 	if !cr.PreferIngress() {
-		status.AdminUrl = fmt.Sprintf("%v.%v.svc.cluster.local:%d", service.Name, cr.Namespace,
+		protocol := "http"
+		if cr.Spec.Config["server"] != nil && cr.Spec.Config["server"]["protocol"] != "" {
```

Comment:
can `cr.Spec.Config` be nil?

Reply:
I think it shouldn't be possible, because CRs cannot be deployed without `config` field:
`The Grafana "grafana-a" is invalid: spec.config: Required value`
But if you feel like it's needed, I could certainly add it.

Btw, discovered that CRs like this lead to operator crash:
```
---
apiVersion: grafana.integreatly.org/v1beta1
kind: Grafana
metadata:
  name: grafana-a
  labels:
    dashboards: "grafana-a"
spec:
```

```
---
apiVersion: grafana.integreatly.org/v1beta1
kind: Grafana
metadata:
  name: grafana-a
  labels:
    dashboards: "grafana-a"
```

Reply:
Added nil check for cr.Spec.Config, moved the logic to a helper function and added tests for it.